Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -34,6 +34,18 @@ public extension TensorFlowScalar {
 }
 
 public extension Tensor {
+    /// Helper function that assess if `axis` is in the range `[-rank, rank)`, where `rank` is the rank of
+    /// the provided tensors.
+    @usableFromInline
+    internal func preconditionAxis(_ axis: Int) {
```

Comment:
This is a helper function to validate that the axis is in the range of -rank,rank. I don’t think it would add much to the public API. Although, I would need to check how useful will be in other modules.